src/openms/source/ANALYSIS/OPENSWATH/IonMobilityScoring.cpp

```diff
@@ -88,35 +72,54 @@ namespace OpenMS
     double max_int = 0;
     for (Size k = 0; k < im_grid.size(); k++)
     {
+      MobilityPeak1D mobi_peak;
       // In each iteration, the IM value of pr_it should be equal to or
       // larger than the master container. If it is equal, we add the current
       // data point, if it is larger we add zero and advance the counter k.
-      if (pr_it != profile.end() && fabs(pr_it->im - im_grid[k] ) < eps*10)
+      if (pr_it != profile.end() && fabs(pr_it->getMobility() - im_grid[k] ) < eps*10)
       {
-        al_int_values.push_back(pr_it->intensity);
-        al_im_values.push_back(pr_it->im);
+        mobi_peak.setIntensity(pr_it->getIntensity());
+        mobi_peak.setMobility(pr_it->getMobility());
+
         ++pr_it;
       }
       else
       {
-        al_int_values.push_back(0.0);
-        al_im_values.push_back( im_grid[k] );
+        mobi_peak.setIntensity(0.0);
+        mobi_peak.setMobility(im_grid[k]);
       }
       // OPENMS_LOG_DEBUG << "grid position " << im_grid[k] << " profile position " << pr_it->first << std::endl;
 
       // check that we did not advance past
-      if (pr_it != profile.end() && (im_grid[k] - pr_it->im) > eps*10)
+      if (pr_it != profile.end() && (im_grid[k] - pr_it->getMobility()) > eps*10)
       {
-        std::cout << " This should never happen, pr_it has advanced past the master container: " << im_grid[k]  << "  / " <<  pr_it->im  << std::endl;
+        std::cout << " This should never happen, pr_it has advanced past the master container: " << im_grid[k]  << "  / " <<  pr_it->getMobility()  << std::endl;
         throw Exception::OutOfRange(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION);
       }
 
       // collect maxima
-      if (pr_it != profile.end() && pr_it->intensity > max_int)
+      if (pr_it != profile.end() && pr_it->getIntensity() > max_int)
       {
-        max_int = pr_it->intensity;
+        max_int = pr_it->getIntensity();
         max_peak_idx = k;
       }
+
+      aligned_profile.push_back(mobi_peak);
+    }
+  }
+
+  void IonMobilityScoring::extractIntensities(const std::vector< Mobilogram >& mobilograms,
+                                              std::vector<std::vector<double>>& int_values)
+  {
+    int_values.clear();
+    int_values.reserve(mobilograms.size());
+
+    for (const auto& mobilogram : mobilograms)
+    {
+      std::vector<double> mobility_int;
+      mobility_int.reserve(mobilogram.size());
+      for (const auto & k : mobilogram) mobility_int.push_back(k.getIntensity());
+      int_values.emplace_back(std::move(mobility_int));
```

Comment:
works.
If you want to tune performance a bit, then use `resize()` for inner and outer vector and work in-place.